python/private/patch_whl.py

```diff
@@ -0,0 +1,180 @@
+"""
+Regenerate a whl file after patching and cleanup the patched contents.
```

Comment:
Is there a reason to reimplement this ourselves rather than using something like:
* https://github.com/pypa/wheel/blob/main/src/wheel/cli/pack.py

Reply:
I will explain my reasons, but some of them might be invalid anymore.

1. We need to move files from one directory to another because the `repository_ctx.patch` can only apply patches to the root directory. I could not find a cross-platform way to `list all files` and move particular ones to a separate dir.
2. Having a zero-dependencies script is a nice thing for writing this helper, because we are dealing with repository rules. Maybe this is actually an actual smell from the fact that we are doing it in the `repository_ctx`.
3. I opted in for reusing the `wheelmaker.py` that we already have in the repo, this way we can have consistent output.

I think I could use the function you've linked, but I would prefer to retain this `RECORD` diffing part for bookkeeping as to what actually is patched. This script has a nice property that it does not change anything that exists in the directory and just zips all of the files ensuring that the `RECORD` file is consistent.